plugins/modules/packaging/language/gem.py

```diff
@@ -213,32 +239,31 @@ def install(module):
         return
 
     ver = get_rubygems_version(module)
-    if ver:
-        major = ver[0]
-    else:
-        major = None
 
     cmd = get_rubygems_path(module)
     cmd.append('install')
+    cmd.extend(common_opts(module))
```

Comment:
The only improvement I can think of right now is to only call `get_rubygems_version` once and store the result in a variable that's passed around. (Or cache it in that function.) Besides that, looks good to me!

Reply:
Is there a common practice on caching?

Reply:
There are many different ways to do this. One simple approach is abusing the fact that Python functions are objects that can store additional data: https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_internal/docker_util.py#L105-L127

Reply:
Simple result caching decorator for future use https://gist.github.com/giner/78f8f631e5f41ec2a3063fb9498a462a

Reply:
It probably makes sense to also support `**kwargs` in the wrapper. Also note that the stdlib [functools](https://docs.python.org/3/library/functools.html) has multiple caches that are also useful - unfortunately they are Python 3 only, so they aren't really usable in Ansible modules :)